src/pkg/caendr/caendr/services/tools/submit.py

```diff
@@ -461,6 +461,12 @@ def parse_data(cls, data, delimiter='\t'):
     # Compute data hash using entire file
     data_hash = get_file_hash(local_path, length=32)
 
+    # Open the file and extract the trait name from the header row
+    with open(local_path, 'r') as f:
+      csv_reader = csv.reader(f, delimiter=delimiter)
+      header_row = next(csv_reader)
+      data['trait'] = header_row[1]
```

Comment:
Can we assert that the trait looks like a trait? Especially for mal-formed files.

Reply:
We'd have to ask Erik if there's any expected format. As far as I can tell, a trait can be any arbitrary string (barring tabs & newlines), but we might(?) want to filter out special characters...